Training/Training/Reversal.c

```diff
@@ -0,0 +1,163 @@
+// ------------------------------------------------------------------------------------------------
+// Assignment4 ~ A training program for new joiners at Metamation, Batch - Sept 2024.
+// Copyright (c) Gokul TS @Metamation India.
+// ------------------------------------------------------------------
+// Reversal.c
+// C program to Check Palindrome or not
+// ------------------------------------------------------------------------------------------------
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+#include <ctype.h>
+
+#define ANSI_RESET_ALL "\x1b[0m"
+#define ANSI_COLOR_RED "\x1b[31m"
+#define ANSI_COLOR_GREEN "\x1b[32m"
+#define ANSI_COLOR_BLUE "\x1b[36m"
+#define ANSI_COLOR_MAGENTA "\x1b[35m"
+
+#define MAX_DIGIT 13
+#define MAX_CHAR 52
+#define PALINDROME 1
+#define NOT_PALINDROME 0
+#define INVALID -1
+
+/// <summary>Function to check given number is palindrome or not </summary>
+int NumIsPalindrome (int num, int type);
+/// <summary>Function to check given string is palindrome or not </summary>
+int StrIsPalindrome (char* phrase, int type);
+/// <summary>Function to get integer input from user</summary>
+void IntegerInput (int option);
+/// <summary>Function to get string input from user</summary>
+void StringInput (int option);
+/// <summary>Function to check test cases</summary>
+void OutputCheck ();
+/// <summary>Function to print result</summary>
+void PrintResult (int value);
+/// <summary>Clears the content on screen </summary>
+void ClrScreen ();
+
+int main () {
+   while (1) {
+      int count = 0;
+      printf ("1-PALINDROME CHECKER\n2-REVERSE A NUMBER\n3-TEST CASES\n4-CLEAR SCREEN\n5-EXIT\nEnter an option:");
+      char option = fgetc (stdin);
+      int num = option;
+      if (num != 10) while (getchar () != '\n') count++;
+      if (count == 0 && (num < 54 && num > 48)) {
+         switch (num) {
+            case '1':
+               StringInput (option);
+               break;
+            case '2':
+               IntegerInput (option);
+               break;
+            case '3':
+               OutputCheck ();
+               break;
+            case '4':
+               ClrScreen ();
+               break;
+            case '5':
+               return 0;
+               break;
+         }
+      } else printf (ANSI_COLOR_RED"Please enter number between 1-5\n\n"ANSI_RESET_ALL);
+   }
+   return 0;
+}
+
+int NumIsPalindrome (int num, int type) {
+   bool isNegative = false;
+   if (num < 0) {
+      num = -num;
+      isNegative = true;
+   }
+   int org = num;
+   long long int rev = 0;
+   while (num != 0) {
+      rev = rev * 10 + num % 10;
+      num /= 10;
+   }
+   if (type == 1) printf ("Output:%lld\n", rev);

```

Comment:
I/O functions should not be present in the functionality. You may return the result and print it outside.